centreon/www/class/centreonStatistics.class.php

```diff
@@ -407,6 +407,48 @@ public function getAccData(): array
         return $data;
     }
 
+    /**
+     * Get poller/agent configuration data.
+     *
+     * @return array
+     */
+    public function getAgentConfigurationData(): array
+    {
+        $data = ['total' => 0];
+
+        $statement = $this->dbConfig->executeQuery(
+            <<<'SQL'
+                SELECT count(id)
+                FROM agent_configuration ac
+                SQL
+        );
+
+        $total = $statement->fetchColumn();
+
+        if ($total === false) {
+            return $data;
+        }
+
+        $data['total'] = $total;
+
+        $statement = $this->dbConfig->executeQuery(
+            <<<'SQL'
+                SELECT type, count(id) as nb_ac_per_type, count(rel.poller_id) as nb_poller_per_type
+                FROM agent_configuration ac
+                JOIN ac_poller_relation rel
+                    ON ac.id = rel.ac_id
+                GROUP BY type
+                SQL
+        );
+
+        foreach ($statement->fetchAll(\PDO::FETCH_ASSOC) as $row) {
+            $data[$row['type']]['configuration'] = $row['nb_ac_per_type'];
+            $data[$row['type']]['pollers'] = $row['nb_poller_per_type'];
+        }
```

Comment:
Now, with CentreonDB, you must not to use PDO native, but the metods of CentreonDB ;) 
Also, a variable to store the result of fetchAll before the loop is more optimized :)
```suggestion
        $acAndPollerPerType = $this->dbConfig->fetchAll($statement);
        foreach ($acAndPollerPerType as $row) {
            $data[$row['type']]['configuration'] = $row['nb_ac_per_type'];
            $data[$row['type']]['pollers'] = $row['nb_poller_per_type'];
        }
```

Reply:
ok